bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/api/esp32/WlanThermoEsp32CommandHandler.java

```diff
@@ -0,0 +1,298 @@
+                            if (command == OnOffType.ON) {
+                                value = BigInteger.valueOf(channel.get(channelId).getAlarm()).setBit(0);
+                            } else {
+                                value = BigInteger.valueOf(channel.get(channelId).getAlarm()).clearBit(0);
+                            }
+                            channel.get(channelId).setAlarm(value.intValue());
+                            success = true;
+                        }
+                        break;
+                    case CHANNEL_COLOR_NAME:
+                        if (command instanceof StringType) {
+                            channel.get(channelId).setColor(UtilEsp32.toHex(((StringType) command).toString()));
+                            success = true;
+                        }
+                        break;
+                }
+            }
+        } else if (channelUID.getId().equals("pit1")) {
+            if (data.getPitmaster() != null && data.getPitmaster().getPm() != null
+                    && data.getPitmaster().getPm().size() > 0) {
+                Pm pm = data.getPitmaster().getPm().get(0);
+                switch (channelUID.getIdWithoutGroup()) {
+                    case CHANNEL_PITMASTER_CHANNEL_ID:
+                        pm.setChannel(((QuantityType<?>) command).intValue());
+                        success = true;
+                        break;
+                    case CHANNEL_PITMASTER_PIDPROFILE:
+                        pm.setPid(((QuantityType<?>) command).intValue());
+                        success = true;
+                        break;
+                    case CHANNEL_PITMASTER_SETPOINT:
+                        pm.setSet(((QuantityType<?>) command).doubleValue());
+                        success = true;
+                        break;
+                    case CHANNEL_PITMASTER_STATE:
+                        String state = ((StringType) command).toString();
+                        if (state.equalsIgnoreCase("off") || state.equalsIgnoreCase("manual")
+                                || state.equalsIgnoreCase("auto")) {
+                            pm.setTyp(state);
+                            success = true;
+                        }
+                }
+            }
+        }
+        return success;
+    }
+
+    public String getTrigger(ChannelUID channelUID, Data data) {
+        String trigger = null;
+        String groupId = channelUID.getGroupId();
+        if (groupId == null || data == null) {
+            return null;
+        }
+        List<Channel> channel = data.getChannel();
+        if (channelUID.getId().startsWith("channel")) {
+            int channelId = Integer.parseInt(groupId.substring("channel".length())) - 1;
+            if (channel.size() > 0 && channelId < channel.size()) {
+                if (CHANNEL_ALARM_OPENHAB.equals(channelUID.getIdWithoutGroup())) {
+                    if (channel.get(channelId).getTemp() != 999) {
```

Comment:
Using the `&&` operator would reduce the amount of nested indentation going on here.

Reply:
Agree, but it wouldn't make the code clearer or better to understand. I tried to structure the different functions in the command handler all in the same way, so one can easily figure out where e.g. new Triggers have to be added.

In the end, the compiler is optimizing it anyways, so I'd plead for leaving it more readable.

Reply:
Fair enough.